### PR TITLE
add settings api for handling invalid config

### PR DESCRIPTION
**Pull request by ilackarms**
expose an api in the Gloo Settings object to configure options for handling invalid user config.

this will be used to configure sanitizers in Gloo (introduced in #1400 )

### CODE REVIEW

**Comment by ilackarms** [comment]
/kick

build  flake 

```
Step #13 - "compile": failed to start io pipe copy: containerd-shim: opening /var/run/docker/containerd/ef4738260a1a288001733ff68f1097e8433235f56f5537f4e0f79a30777a8dbe/init-stdout failed: context deadline exceeded: unknown

```

**Comment by ilackarms** [comment]
/kick

```
Step #11 - "test": • Failure in Spec Setup (BeforeEach) [1.760 seconds]
Step #11 - "test": StatusSyncer [BeforeEach] updates kube ingresses with endpoints from the service 
Step #11 - "test": /workspace/gopath/src/github.com/solo-io/gloo/projects/clusteringress/pkg/status/status_syncer_test.go:57
Step #11 - "test": 
Step #11 - "test":   Unexpected error:
Step #11 - "test":       <*url.Error | 0xc00067d350>: {
Step #11 - "test":           Op: "Post",
Step #11 - "test":           URL: "https://35.232.0.2/api/v1/namespaces",
Step #11 - "test":           Err: {
Step #11 - "test":               Op: "dial",
Step #11 - "test":               Net: "tcp",
Step #11 - "test":               Source: nil,
Step #11 - "test":               Addr: {
Step #11 - "test":                   IP: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 255, 255, 35, 232, 0, 2],
Step #11 - "test":                   Port: 443,
Step #11 - "test":                   Zone: "",
Step #11 - "test":               },
Step #11 - "test":               Err: {Syscall: "connect", Err: 0x6f},
Step #11 - "test":           },
Step #11 - "test":       }
Step #11 - "test":       Post https://35.232.0.2/api/v1/namespaces: dial tcp 35.232.0.2:443: connect: connection refused
Step #11 - "test":   occurred
Step #11 - "test": 
Step #11 - "test":   /workspace/gopath/src/github.com/solo-io/gloo/projects/clusteringress/pkg/status/status_syncer_test.go:51
Step #11 - "test": 

```

**Comment by ilackarms** [comment]
/kick 


Step #11 - "test": /workspace/gopath/src/github.com/solo-io/gloo/test/e2e/access_log_test.go:171
Step #11 - "test": 
Step #11 - "test":   Expected
Step #11 - "test":       <string>: 
Step #11 - "test":   to equal
Step #11 - "test":       <string>: local-1_default
Step #11 - "test": 
Step #11 - "test":   /workspace/gopath/src/github.com/solo-io/gloo/test/e2e/access_log_test.go:202
Step #11 - "test":

**Comment by ilackarms** [comment]
/kick

there appears to be a flake with knative test (i reproduced locally) where the knative `routes` crd was somehow missing, preventing knative from working properly

File: install/helm/gloo/generate/values.go [hunk 1]
```diff
@@ -92,12 +92,19 @@ type KnativeProxy struct {
...
+	WriteNamespace      string               `json:"writeNamespace,omitempty" desc:"namespace where intermediary CRDs will be written to, e.g. Upstreams written by Gloo Discovery."`
+	Integrations        *Integrations        `json:"integrations,omitempty"`
+	Create              bool                 `json:"create" desc:"create a Settings CRD which provides bootstrap configuration to Gloo controllers"`
+	Extensions          interface{}          `json:"extensions,omitempty"`
+	SingleNamespace     bool                 `json:"singleNamespace" desc:"Enable to use install namespace as WatchNamespace and WriteNamespace"`
+	InvalidConfigPolicy *InvalidConfigPolicy `json:"invalidConfigPolicy" desc:"Define policies for Gloo to handle invalid configuration"`
+}
+
+type InvalidConfigPolicy struct {
+	ReplaceInvalidRoutes     bool   `json:"replaceInvalidRoutes,omitempty"`
```
Comment (marcogschmidt) [inline install/helm/gloo/generate/values.go]: Can you add a description tag to these attributes as well? Maybe a succinct version of the ones on the proto.

File: projects/gloo/api/v1/settings.proto [hunk 2]
```diff
@@ -326,6 +326,28 @@ message GlooOptions {
     }
 
     AWSOptions aws_options = 5;
+
+    // policy for how Gloo should handle invalid config
```
Comment (marcogschmidt) [inline projects/gloo/api/v1/settings.proto]: Nit: capitalize as appropriate to be consistent with the other comments on this proto. Also some of the comments need a bit of rewording, e.g. "set replaceInvalidRoutes to `true`, Gloo remove any routes from..."

**Pull request merged**